overlord/snapstate/snapstate_test.go

```diff
@@ -5772,7 +5811,34 @@ func (s *snapmgrTestSuite) TestGadgetDefaults(c *C) {
 	c.Assert(runHook.Kind(), Equals, "run-hook")
 	err = runHook.Get("hook-context", &m)
 	c.Assert(err, IsNil)
-	c.Assert(m["patch"], DeepEquals, map[string]interface{}{"key": "value"})
+	c.Assert(m, DeepEquals, map[string]interface{}{"use-defaults": true})
+}
+
+func (s *snapmgrTestSuite) TestInstallPathSkipConfigure(c *C) {
+	r := release.MockOnClassic(false)
+	defer r()
+	dirs.SetRootDir(c.MkDir())
+	defer dirs.SetRootDir("")
+
+	// using MockSnap, we want to read the bits on disk
+	snapstate.MockReadInfo(snap.ReadInfo)
+
+	s.state.Lock()
+	defer s.state.Unlock()
+
+	s.prepareGadget(c)
+
+	snapPath := makeTestSnap(c, "name: some-snap\nversion: 1.0")
+
+	ts, err := snapstate.InstallPath(s.state, &snap.SideInfo{RealName: "some-snap", SnapID: "some-snap-id", Revision: snap.R(1)}, snapPath, "edge", snapstate.Flags{SkipConfigure: true})
```

Comment:
Hm, this is confusing: `snapstate.Flags{SkipConfigure: true}` and then we test `c.Check(snapsup.Flags.SkipConfigure, Equals, false)` - maybe a small comment ?